server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -191,14 +190,7 @@ public Response doPost(
         queryId = UUID.randomUUID().toString();
         query = query.withId(queryId);
       }
-      if (query.getContextValue(QueryContextKeys.TIMEOUT) == null) {
```

Comment:
Sorry @jihoonson, I wasn't clear. For backwards compatibility reasons the query context parameter should stay milliseconds. Since `druid.server.http.defaultQueryTimeout` is new it could either be milliseconds or a period.